usaspending_api/transactions/delta_models/transaction_fpds.py

```diff
@@ -363,6 +363,7 @@
     TransactionColumn("modification_number", "award_modification_amendme", "STRING"),
     TransactionColumn("non_federal_funding_amount", "NULL", "NUMERIC(23, 2)", "literal"),
     TransactionColumn("original_loan_subsidy_cost", "NULL", "NUMERIC(23, 2)", "literal"),
+    # All period_of_performance_* fields seen as: YYYY-MM-DD 00:00:00, so cast works
     TransactionColumn("period_of_performance_current_end_date", "period_of_performance_curr", "DATE", "cast"),
```

Comment:
Although we have only seen `period_of_performance_*` fields as YYYY-MM-DD format in fpds, I'd still put the handling as `parse_string_datetime_to_date` to be more robust against the possibility of seeing other formats in the future.

Reply:
Had second thoughts after I approved.  See below.

Reply:
I added the smart-date-parsing for FPDS too and committed